src/umock_log.c

```diff
@@ -8,6 +8,9 @@
 #include "umock_log.h"
 #include "umock_c_internal.h"
 
+#if defined(__GNUC__) || defined(__clang__)
+#pragma GCC diagnostic ignored "-Wformat-nonliteral"

```

Comment:
You should not ignore the warning, but actually tell gcc to validate the args with

[https://gcc.gnu.org/onlinedocs/gcc-3.1/gcc/Function-Attributes.html](https://gcc.gnu.org/onlinedocs/gcc-3.1/gcc/Function-Attributes.html)

__attribute__ ((format (printf, 1, 2)));

Or something like that.

Reply:
Fixed. Thank you for your suggestion. Worked well.